src/arrayMethodJoin.js

```diff
@@ -3,9 +3,28 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let string = '';
+    let insert = '';
```

Comment:
you may declare it into the loop and this empty variable will be created on every iteration